packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -28,10 +28,49 @@ export function PreviewIndicator({ children }) {
 }
 
 let buildId
+let pageData
+let latestCheckedBuild
 
 export default function Indicator() {
   const [buildInfo, setBuildInfo] = useState()
 
+  async function fetchPageData() {
+    const urlHostString = window.location.origin
+    const pathAdjustment =
+      window.location.pathname === `/` ? `/index` : window.location.pathname
+
+    const url = `${urlHostString}/page-data${pathAdjustment}/page-data.json`
+
+    const resp = await fetch(url)
+    const data = await resp.text()
+
+    return data
+  }
+
+  useEffect(() => {
+    // declare the data fetching function
+    const fetchData = async () => {
+      const data = await fetchPageData()
+      pageData = data
+    }
+
+    // call the function
+    fetchData()
+      // make sure to catch any error
+      .catch(console.error)
```

Comment:
```suggestion
    const fetchData = async () => {
      const data = await fetchPageData()
      pageData = data
    }

    fetchData()
      .catch(console.error)
```
What happens if fetching data errors? Do we want to retry? Do we want to put the indicator in a failed state? Could it mean that the page the user is on got deleted?

Reply:
I'd maybe name these functions a more descriptively as well. I could see down the road not knowing what kind of data `fetchData` grabs

Reply:
Wondering if maybe storing a hash of the page data instead of the entire page data will be better?

Reply:
@DanielSLew it'll be more expensive to store a hash since the browser needs to create the hash

Reply:
Holding the page data in browser memory should be fine 👍